beacon-chain/db/filesystem/layout_by_epoch.go

```diff
@@ -0,0 +1,193 @@
+package filesystem
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	"github.com/spf13/afero"
+)
+
+type periodicEpochLayout struct {
+	fs     afero.Fs
+	cache  *blobStorageCache
+	pruner *blobPruner
+}
+
+var _ fsLayout = &periodicEpochLayout{}
+
+func newPeriodicEpochLayout(fs afero.Fs, cache *blobStorageCache, pruner *blobPruner) fsLayout {
+	l := &periodicEpochLayout{fs: fs, cache: cache, pruner: pruner}
+	return l
+}
+
+func (l *periodicEpochLayout) notify(ident blobIdent) error {
+	if err := l.cache.ensure(ident); err != nil {
+		return err
+	}
+	l.pruner.notify(ident.epoch, l)
+	return nil
+}
+
+// If before == 0, it won't be used as a filter and all idents will be returned.
```

Comment:
Why this comment?
IMO this comment applies for `isBeforePeriod` and `isBeforeEpoch`, not to `isRootDir` nor `isSszFile`.

Reply:
It applies to how the returned iterator behaves, overall. My thought was that since this is a method on the layout, it is more likely for someone touching the higher level storage code to see this method in the methodset and make an assumption that `before` would behave a certain way, so I wanted to make that explicit.